lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -838,6 +1160,27 @@ bool InferenceTypeUpdate::updateValue(Value value) {
   if (!type)
     return false;
 
+  // If this is an operation that does not generate any free variables that are
+  // determined during width inference, simply update the value type based on
+  // the operation arguments.
+  if (auto op = dyn_cast_or_null<InferTypeOpInterface>(value.getDefiningOp())) {
+    SmallVector<Type, 2> types;
+    auto res =
+        op.inferReturnTypes(op->getContext(), op->getLoc(), op->getOperands(),
+                            op->getAttrDictionary(), op->getRegions(), types);
+    if (failed(res)) {
+      anyFailed = true;
+      return false;
+    }
+    assert(types.size() == 1 && "all ops of interest must have single result");
+    LLVM_DEBUG(llvm::dbgs()
+               << "Inferring " << op << " as " << types[0] << "\n");
+    for (auto it : llvm::zip(op->getResults(), types)) {
```

Comment:
I realize this won't be true for long, but you did just assert there was only one result.

Reply:
Oops my bad. The update happens in a loop anyway, so the assert is not even needed. I changed it to check that the number of inferred results matches the number of results on the operation.